web/src/pages/AdminPage/UserEditor.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    TextField,
+} from '@mui/material'
+import * as React from 'react'
+import { useEffect, useState } from 'react'
+import { User, UserUpdateRequest } from '../../model/user'
+
+interface UserEditorProps {
+    user: User | null
```

Comment:
In JS `undefined` tends to be used over `null` (easier to work with). It just a nitpick tho, feel free to ignore.

You could then just:
```ts
 user?: User
 ```